jobs/vault/templates/bin/post-start

```diff
@@ -6,6 +6,7 @@ if [[ "$(cat "${JOB_DIR}/data/unseal_keys" | wc -l)" -gt 2 ]]; then
   cat "${JOB_DIR}/data/unseal_keys" | \
     while read key
     do
-      vault unseal ${key}
+      key=$(echo "$key" | tr -d '[:space:]')
```

Comment:
Good call on the space trimming. With regard to the unseal and need for `VAULT_SKIP_VERIFY` as noted in your comment above I'm wondering if it might make sense to run `VAULT_SKIP_VERIFY=1 vault unseal ${key}` here since this script will always be running locally.

Reply:
I was thinking of refactoring properties.sh.erb to set VAULT_SKIP_VERIFY, and realized that is against the [recommendations](https://www.vaultproject.io/docs/commands/index.html#vault_skip_verify).

We use private CA's so we have the flexibility of IP SAN, and I can see other people may not have that option. 

How about we use a new attribute in the `spec` file, like:
```yaml
properties:
  vault.tls_skip_verify:
    description: skip ssl CA cert and CN validation when unseal the vault locally during post-start
    default: false

  vault.config:
    description: HCL string literal representing the full Vault configuration, will take precedence over any other configuration properties
```

Of course I will refactor properties.sh.ejb to parse it.

Reply:
If we passed the VAULT_ADDR to use (or tried to parse it from the config, probably too brittle) then we could specify a valid address given the TLS cert and maybe not need skip verify at all?

Reply:
Something like `vault.local_addr` `the Vault address local operations like the post-script for unsealing will use for communicating to Vault` it could default to 127.0.0.1 which would mean folks that already had the IP SAN for local wouldn't need to do anything special.

Reply:
The code you noted in #59 sets up the properties that the Vault CLI installed on the instance will use. Because `vault_addr=p("vault.address", "")` always returns the empty string, the `VAULT_ADDR` environment variable is [set to the empty string](https://github.com/cloudfoundry-community/vault-boshrelease/blob/master/jobs/vault/templates/data/properties.sh.erb#L15).

This `properties` file gets [sourced by a helper script](https://github.com/cloudfoundry-community/vault-boshrelease/blob/master/jobs/vault/templates/helpers/ctl_setup.sh#L25) that is [sourced in `post-start`](https://github.com/cloudfoundry-community/vault-boshrelease/blob/master/jobs/vault/templates/bin/post-start#L3) so when we're running vault commands in `post-start` because Vault's default behavior is to use `127.0.0.1` we get a TLS error unless our certificate has an IP SAN of 127.0.0.1. If we allow consumers set the `VAULT_ADDR` that gets exported,  then we prevent the TLS error as folks will be able to pass in whatever address routes to the vault instance and is present in their cert.

There's some risk there in that connecting to something other than `127.0.0.1` gives no guarantee of where you're sending the unseal commands. It could be possible for that value to be set maliciously and allow the unseal keys to be stolen by some other server. In either case I think [we end up breaking the full Vault security model](https://www.vaultproject.io/docs/commands/index.html#vault_skip_verify). Maybe what needs to happen is that we provide an environment variable pass through like

Reply:
Hi, Matt, makes sense. I will update the P/R later today. I will also include some blurb in the README.